### DISCUSSION TITLE
Stepping back from Tablesaw

### DISCUSSION THREAD

**Discussion by lwhite1** [jtablesaw/tablesaw#1233] [General]
Hi All, 

As I'm sure some have noticed, PRs are no longer getting merged in a timely fashion and issues are not being resolved as they once were. This is largely because I've reassessed how I spend my time, and find I'm no longer sufficiently interested in the project to devote time to it.  I'm sorry for any negative impact this causes. 

Over the years, there have been innumerable contributions by many people -- without which Tablesaw would never have been unusable. At this point, however, no one who's made extensive contributions to the project still works actively on it (with the exception of Ben, who does keep merging security fixes and occasional other contributed code). This leaves the project without an obvious 'heir' to continue to move it forward. 

If anyone wants to either take over the project or create a maintained fork, please let me know. 

I would to thank once again everyone who has contributed to Tablesaw, and everyone who has used it.  

Best.
Larry

**Comment by s1ny1998** [comment]
Hi Larry,

I hope you're doing well.

I'm Sanmitra, a senior software developer with experience in full stack development. I've been following Tablesaw and am very interested in contributing by helping to maintain the project.

However, I recognize that every project has its unique challenges and nuances. Therefore, I would greatly appreciate some guidance on how to proceed with maintaining or managing the project effectively. Specifically, I would like to understand:

 -  Any immediate priorities or tasks that need attention.
 -  The current workflow and development practices that the team follows.
 -  How best to communicate and collaborate with the team and community.
 -  Any additional documentation or resources that would be helpful for a smooth transition.

Thank you for considering my request. I look forward to your response and the opportunity to support the project.

Regards,
Sanmitra

**Comment by lwhite1** [comment]
Hi, See below.

On Tue, Jun 11, 2024 at 7:53 AM Sanmitra Nagaraj ***@***.***>
wrote:

> Hi Larry,
>
> I hope you're doing well.
>
> I'm Sanmitra, a senior software developer with experience in full stack
> development. I've been following Tablesaw and am very interested in
> contributing by helping to maintain the project.
>
> However, I recognize that every project has its unique challenges and
> nuances. Therefore, I would greatly appreciate some guidance on how to
> proceed with maintaining or managing the project effectively. Specifically,
> I would like to understand:
>
>    - Any immediate priorities or tasks that need attention.
>
> I don't think there are any immediate priorities.   The biggest issue is
to get the new branch working or fall-back to supporting the old branch.
The effort to get the new branch working is probably significant.

>
>    - The current workflow and development practices that the team follows.
>
> These are documented on the site. See the contributing guide
<https://jtablesaw.github.io/tablesaw/contributing>.  The most complex part
is the document source code, but if the system doesn't change much it may
be best to simplify this and maintain docs manually. Overall, it may be
best to start a new fork. We could link to that fork as the new 'official'
version.

>
>    - How best to communicate and collaborate with the team and community.
>
> The best way is to use discussions and issues. It wouldn't hurt to set an
alert on stack overflow for Tablesaw so you can track and/or help with
questions asked there.

>
>    - Any additional documentation or resources that would be helpful for
>    a smooth transition.
>
> Thank you for considering my request. I look forward to your response and
> the opportunity to support the project.
>
Thank you.

**Comment by ccleva** [comment]
Hi @lwhite1,

I started working on a maintenance fork of Tablesaw. Is it still your preferred solution to move the project forward ?

It would be released under a different organization though, let me know if this is an issue.

Thanks,
Best

**Reply by benmccann** [reply]
I would prefer not to fork the project and community. I'm afraid I only watch PRs and not issues or discussion, so didn't see this discussion since I wasn't tagged in it. I don't consider this project to be unmaintained though I admit I've been a bit slow to review PRs lately. If there are changes from your maintenance fork that you'd like to see landed here, let's work together to make that happen

**Reply by ccleva** [reply]
Sure, as you prefer. The maintenance fork is based on the java8 version (0.43.1), integrates bug fixes and dependencies upgrade from the master branch, plus additional fixes. I will make PRs on master for the additional fixes, but how do you want to proceed for the java8 branch ?

**Reply by benmccann** [reply]
Ah, I see Larry upgraded from Java 8 to Java 11 right after cutting the last release: https://github.com/jtablesaw/tablesaw/pull/1095

Do you still need Java 8 support? If so, I wonder if there's anything in this library that is actually requiring Java 11. Perhaps we could just set it back to Java 8 if needed?

I've mostly been doing JS lately and haven't kept up with the Java ecosystem much since the Java 8 days to be honest. Would dropping Java 8 be disruptive for folks? I see Java 11 was released in September 2018, so requiring it seems reasonable. I'd be inclined to just bump the version number and drop Java 8 support, but don't have any real feelings about it one way or the other at the moment.

**Reply by ccleva** [reply]
I don't need java 8 support myself and would happily migrate to 11, it was just less disruptive to start from this. Some of the new PRs might use java 11 specifics that will require a little code cleanup if you want to revert to 8.

I am also undecided about java 8, several projects require 11 already, but java 8 is still in LTS.

**Reply by benmccann** [reply]
It looks like there is at least a little syntax in `master` that requires Java 11:
https://github.com/jtablesaw/tablesaw/blame/6daa05acfe716e140984896873cc27757677405b/core/src/main/java/tech/tablesaw/columns/strings/ByteDictionaryMap.java#L404

It looks like Java 8 is dropped from "Premier Support", but not "Extended Support": https://endoflife.date/oracle-jdk

Anyway, I say we just leave it as Java 11 and bump the version number. If someone desperately needs Java 8 support they're free to do the leg work of figuring out what would need to change and proposing it and we can evaluate at that time if the extra syntax is worth the cost

**Comment by lwhite1** [comment]
That sounds great.  Best of luck.

**Comment by benmccann** [comment]
Thank you for this project Larry! It's been great working with you on it. I'm afraid I didn't see this discussion as I wasn't tagged in it, but I'll try to keep things alive here as well as I can if folks would like to help out. I believe @ryancerf was also contributing for awhile.

It looks like the main need right now is simply to cut a new release (https://github.com/jtablesaw/tablesaw/issues/1251) as there are PRs I've merged that folks would like to take advantage of, so I'll go ahead and do that
